packages/pigeon/lib/dart_generator.dart

```diff
@@ -107,8 +108,28 @@ String _makeGenericCastCall(TypeDeclaration type, String nullTag) {
       : '';
 }
 
-String _getArgumentName(NamedType field) =>
-    field.name.isEmpty ? 'arg' : field.name;
+String _getArgumentName(int count, NamedType field) =>
```

Comment:
In my mind they are both required data for the calculation.  The only reason I'll prefer one order over another is if I'm trying to mimic object oriented programming where it isn't available (like C) or if it's a higher order function I might put the function parameter at the end to make it a bit cleaner when passing in anonymous closures.  Otherwise it's just not a thought.

Reply:
If you don't care, then can we swap it? I consider it pretty standard practice to have `getSomething(primarySource, fallback)` rather than `getSomething(fallback, primarySource)`.

Reply:
I just went to do it and there is another reason that they are in this order that I didn't see, it's the order required by `indexMap`.  If i change the order I'll have to map a lambda function just to swizzle the order of parameters.